internal/services/azurestackhci/stack_hci_logical_network_resource_test.go

```diff
@@ -0,0 +1,270 @@
+package azurestackhci_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/logicalnetworks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type StackHCILogicalNetworkResource struct{}
+
+// https://learn.microsoft.com/en-us/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites
+// The resource can only be created on the customlocation generated after HCI deployment
+const (
+	customLocationIdEnv = "ARM_TEST_STACK_HCI_CUSTOM_LOCATION_ID"
+)
+
+func TestAccStackHCILogicalNetwork_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_logical_network", "test")
+	r := StackHCILogicalNetworkResource{}
+
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skip the test as one or more of below environment variables are not specified: %q", customLocationIdEnv)
+	}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStackHCILogicalNetwork_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_logical_network", "test")
+	r := StackHCILogicalNetworkResource{}
+
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skip the test as one or more of below environment variables are not specified: %q", customLocationIdEnv)
+	}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
```

Comment:
please use `basic` -> `complete` -> `update` -> `basic` to also test add/remove tags.

Reply:
test updated